arch/rv32i/src/lib.rs

```diff
@@ -104,17 +104,17 @@ pub extern "C" fn _start() {
             la a1, {edata}              // a1 = first address after data section in RAM
             la a2, {etext}              // a2 = address of stored data initial values
 
-          data_init_loop:
-            beq  a0, a1, data_init_done // If we have reached the end of the .data
+          200: // data_init_loop
+            beq  a0, a1, 201f           // If we have reached the end of the .data
                                         // section then we are done.
             lw   a3, 0(a2)              // a3 = *a2. Load value from initial values into a3.
             sw   a3, 0(a0)              // *a0 = a3. Store initial value into
                                         // next place in .data.
             addi a0, a0, 4              // a0 = a0 + 4. Increment to next word in memory.
             addi a2, a2, 4              // a2 = a2 + 4. Increment to next word in flash.
-            j data_init_loop            // Continue the loop.
+            j 200b                      // Continue the loop.
```

Comment:
okay, they are all "number plus surprise `f`" except this one which is "number plus surprise `b`"??

I'm going to ask a question that I probably know the dissatisfying answer to: are we sure that label-free is the future of rust, or is there a chance we can hold out and they'll revert or fix this?

Reply:
You can reuse labels! Because why not! https://sourceware.org/binutils/docs/as/Symbol-Names.html#Local-Labels

And, do you like to read discouraging github threads? https://github.com/rust-lang/rust/issues/81088